src/pages/Strategies/hooks/use-get-strategy-proxy-account.ts

```diff
@@ -0,0 +1,92 @@
+import { storageKeyToNthInner } from '@interlay/interbtc-api';
+import { AccountId } from '@polkadot/types/interfaces';
+import { useErrorHandler } from 'react-error-boundary';
+import { useQuery } from 'react-query';
+
+import useAccountId from '@/hooks/use-account-id';
+
+import { StrategyType } from '../types';
+
+interface UseGetStrategyProxyAccountResult {
+  account: AccountId | undefined;
+  isIdentitySet: boolean | undefined;
```

Comment:
What is this for? Could not grasp completely from the code.

Reply:
This contains the information whether proxy account identity was set or not - if it wasn't then we need to set it within deposit extrinsic.